lib/ua_update_messaging.py

```diff
@@ -104,14 +104,47 @@ def _write_esm_service_msg_templates(
     no_pkgs_file: str,
     motd_pkgs_file: str,
     motd_no_pkgs_file: str,
-    no_warranty_file: str,
+    remove_template_files: bool = False,
 ):
+    """Write any related template content for an ESM service.
 
+    If no content is applicable for the current service state, remove
+    all service-related template files.
+
+    :param cfg: UAConfig instance for this environment.
+    :param ent: entitlements.base.UAEntitlement,
+    :param expiry_status: Current ContractExpiryStatus enum for attached VM.
+    :param remaining_days: Int remaining days on contrat, negative when
+        expired.
+    :param *_file: template file names to write.
+
+    :param remove_template files: True when all related template should be
+        removed.
+    """
     pkgs_msg = no_pkgs_msg = motd_pkgs_msg = motd_no_pkgs_msg = ""
-    no_warranty_msg = ""
     tmpl_prefix = ent.name.upper().replace("-", "_")
     tmpl_pkg_count_var = "{{{}_PKG_COUNT}}".format(tmpl_prefix)
     tmpl_pkg_names_var = "{{{}_PACKAGES}}".format(tmpl_prefix)
+    msg_dir = os.path.join(cfg.data_dir, "messages")
+    if remove_template_files:
+        # Purge all template out output messages for this service
+        _write_template_or_remove("", os.path.join(msg_dir, no_pkgs_file))
+        _write_template_or_remove("", os.path.join(msg_dir, pkgs_file))
+        _write_template_or_remove("", os.path.join(msg_dir, motd_no_pkgs_file))
+        _write_template_or_remove("", os.path.join(msg_dir, motd_pkgs_file))
+        return
```

Comment:
When a certain flag is passed to this function it does something very different from what it normally does - it removes all the files instead of going through all messaging logic and potentially creating files. This feels to me like it warrants a separate "remove_all_template_files" function rather than a special case within this function.

Reply:
Great point, I agree with this sentiment and let sloppiness get the better of me in this PR. I separated _remove_msg_templates function which in turn simplifies _write_esm_service_msg_templates